components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/GetAssociatedLocalUserFunctionImpl.java

```diff
@@ -42,9 +44,26 @@ public JsAuthenticatedUser getAssociatedLocalUser(JsAuthenticatedUser federatedU
             }
             return null;
         }
-        String externalSubject = federatedUser.getWrapped().getAuthenticatedSubjectIdentifier();
         String tenantDomain = federatedUser.getWrapped().getTenantDomain();
         String externalIdpName = federatedUser.getWrapped().getFederatedIdPName();
+        String externalSubject = null;
+        try {
+            String userIdClaimURI = Utils.getUserIdClaimURI(externalIdpName, tenantDomain);
+            if (StringUtils.isNotEmpty(userIdClaimURI)) {
+                externalSubject = federatedUser.getWrapped().getUserAttributes().entrySet().stream().filter(
```

Comment:
can federatedUser.getWrapped().getUserAttributes() be null? if so handle null case

Reply:
Addressed in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/75/commits/08aa7f79fa7b1d06d543e6e02029a78df1e2c073